uber/site_sections/preregistration.py

```diff
@@ -162,7 +162,12 @@ def form(self, session, message='', edit_id=None, **params):
                 else:
                     target = group if group.badges else attendee
                     track_type = c.EDITED_PREREG if target.id in self.unpaid_preregs else c.UNPAID_PREREG
-                    self.unpaid_preregs[target.id] = to_sessionized(attendee, group)
+                    if attendee.paid == c.NEED_NOT_PAY:
```

Comment:
Currently in the live system, what happens is:
1. An attendee fills out their badge.
2. Even if that badge is free (e.g., it's for a child under 6), the attendee can add more badges to their cart.
3. If all the badges in their cart are free (...for some reason), the "pay now" button on the index page is replaced with a confirmation button. After clicking this button, the registrations are all processed as free and the attendee is taken to the confirmation page.

What this change (and the accompanying change below) would do is:
1. An attendee fills out their badge.
2. If the attendeee's badge is free, they are taken right to the confirmation page.

We'd really like to keep the cart system in effect -- for one thing, I don't know what would actually happen with this change if someone fills out a paid badge first and then tries to add a free badge to their cart. The current system already behaves correctly now, so we just need to undo this change. (IIRC, this PR was started before we updated the system to handle free badges, which is why this change is here.)